llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -1889,11 +1890,16 @@ MCSection *TargetLoweringObjectFileCOFF::getSectionForJumpTable(
 bool TargetLoweringObjectFileCOFF::shouldPutJumpTableInFunctionSection(
     bool UsesLabelDifference, const Function &F) const {
   if (TM->getTargetTriple().getArch() == Triple::x86_64) {
-    if (!JumpTableInFunctionSection) {
-      // We can always create relative relocations, so use another section
-      // that can be marked non-executable.
-      return false;
+    // Check the module flag (set by frontend when -fjumptable-in-function-section is used)
+    if (const auto *MD = F.getParent()->getModuleFlag("jumptable-in-function-section")) {
```

Comment:
Since the module and Clang flags only effect x64, I recommend prefixing them with `x64`.